TPU: keras support for multiple layer nodes

ctessum opened the pull request:
Hopefully fixes #23659, but I'm having some difficulty testing it.

akuegel committed:
Only convert BackpropFilterConv to depthwise convolution if format is NHWC.

The logic for the conversion has this assumption. This CL makes sure that this
assumption holds, and adds tests for NCHW format.

PiperOrigin-RevId: 225534508

qlzh727 committed:
Fixing tf nightly build.

PiperOrigin-RevId: 225548043

tfboyd committed:
Upgrade to CUDA 10

rohan100jain committed:
Raising an error when different transformations in a dataset pipeline are in different graphs. This is in preparation for the next TF release where this warning would become an error.

PiperOrigin-RevId: 225875718

martinwicke committed:
Add a warning message which happens when contrib is imported.

PiperOrigin-RevId: 225933307

wenxizhu committed:
Bug fix for reduandant transpose removal:
1. this bug is a last minute check-in to replace "CHECK()" with "DCHECK()", however it turns out that "CHECK()" and "DCHECK()" are not functional equivalent;
2. unlike "CHECK()" whose expression will always be excuted, the expression in "DCHECK()" will not be executed in release mode. This causes dangling pointer failures.
3. to fix this bug, we should check the result of the expression instead of the expression itself.

aselle committed:
Merge pull request #24411 from rohan100jain/cherrypicks_CCI04

r1.13-rc0 cherrypick request: Raise warning for dataset pipeline in different graphs

tensorflower-gardener committed:
Update astor version to fix https://github.com/berkerpeksag/astor/issues/86 which we are hitting in Python 3.7

PiperOrigin-RevId: 225966877

aselle committed:
Merge pull request #24432 from goldiegadde/cherrypicks_QP3RO

Fixing tf nightly build.

annarev committed:
Merge pull request #24433 from aselle/cherrypicks_65FMJ

Add a warning message which happens when contrib is imported.

annarev committed:
Update toolchain for arm.

PiperOrigin-RevId: 226075926

annarev committed:
Add k8 to toolchains.

PiperOrigin-RevId: 226087365

annarev committed:
Add k8 to toolchains.

PiperOrigin-RevId: 226087365

fdxmw committed:
Internal change.

PiperOrigin-RevId: 226087680

wenxizhu committed:
Use TF_CHECK_OK() to check:
1. perm_node->attr();
2. AddEdge().

fdxmw committed:
Internal change.

PiperOrigin-RevId: 226094885

aselle committed:
Merge pull request #24436 from goldiegadde/cherrypicks_QP3RO

Update toolchain for arm.

aselle committed:
Merge pull request #24438 from annarev/cherrypicks_V9YYC

1.13.0-rc0 cherry-pick request: Add k8 to toolchains

annarev committed:
Internal change.

PiperOrigin-RevId: 226115035

annarev committed:
Internal change.

PiperOrigin-RevId: 226115035

ebrevdo committed:
[TF port] Disable tests for GetCurrentCpu() on iOS/OS X because MacOS has a __cpuid bug.

PiperOrigin-RevId: 226125575

annarev committed:
Merge pull request #24440 from annarev/cherrypicks_XIW5M

1.13.0-rc0 cherry-pick request: add armebi toolchain

aselle committed:
Update version to 1.13.0-rc0

aselle committed:
Merge pull request #24451 from aselle/update-version

Update version to 1.13.0-rc0

aselle committed:
Handle Python 2 and Python 3 compatibility by forcing to bytes.

This had previously been done but not uniformly. Change to remove
#!/usr/bin/env python in October caused this problems in latest release.

PiperOrigin-RevId: 226221579

aselle committed:
Remove test that is no longer in respository.

PiperOrigin-RevId: 226230363

tensorflower-gardener committed:
Remove second bazel installation

PiperOrigin-RevId: 226233807

aselle committed:
Automated rollback of commit 7b9865971ed38fb3c46aa15c64c6660c50af2d83. Revert #23929.

PiperOrigin-RevId: 226240832

samikama committed:
Initialize only CPU devices in optimize_dataset_op

samikama committed:
Fix clang format

jdduke committed:
Include op defs in the toco converter bindings

This allows conversion of TFLite models with support for select
TF ops via the standard pip installation.

Note that support for *executing* TFLite models with select TF
ops, via Python, will land separately.

PiperOrigin-RevId: 226355228

gunan committed:
Bump max version of bazel to 0.21

PiperOrigin-RevId: 226385826

annarev committed:
Add more toolchain cpus to BUILD file and remove corresponding default
toolchains from CROSSTOOL files.

PiperOrigin-RevId: 226385901

tensorflower-gardener committed:
Internal change

PiperOrigin-RevId: 226386044

goldiegadde committed:
Merge branch 'r1.13' of https://github.com/tensorflow/tensorflow into cherrypicks_QP3RO

aselle committed:
Do not run removed tests.

PiperOrigin-RevId: 226419504

aselle committed:
Merge pull request #24486 from goldiegadde/cherrypicks_QP3RO

Cherrypicks qp3 ro

aselle committed:
Make estimator part of pip install

PiperOrigin-RevId: 226422176

aselle committed:
Merge pull request #24489 from aselle/cherry-pick-4

Misc. build fixes and fixes to toco converter.

aselle committed:
Merge pull request #24498 from aselle/cherry-pick-5

Make estimator part of pip install

jdduke committed:
Enable some smartreply tests in oss

PiperOrigin-RevId: 226525949

stephanwlee committed:
Fix GFile tf_export for v2.

GFile was not properly exposed under io.gfile.GFile.

PiperOrigin-RevId: 226959951

gunan committed:
Upgrade default TF cuda version to cuda 10.

PiperOrigin-RevId: 227180147

gunan committed:
Move default TF builds to cuda 10.

PiperOrigin-RevId: 227187422

perfinion committed:
cloud oauth_client: update for OpenSSL 1.1.0 compatibility

EVP_MD_CTX_cleanup was removed in OpenSSL 1.1.0. There is a call to
EVP_MD_CTX_destroy right after and _destroy will call _cleanup if
required. EVP_MD_CTX_destroy exists in OpenSSL 1.0, 1.1 and BoringSSL so
removing the call to _cleanup works everywhere.

Signed-off-by: Jason Zaman <jason@perfinion.com>

aselle committed:
Merge pull request #24664 from aselle/cherry-pick-20190102

[TF port] Disable tests for GetCurrentCpu() on iOS/OS X because MacOS…

aselle committed:
Make tests all have their own main so initialization works.

PiperOrigin-RevId: 227735303

aselle committed:
Disable on mac a flaky test.

PiperOrigin-RevId: 227746737

aselle committed:
Merge pull request #24689 from aselle/cherry-pick-20190103

Cherry pick build fixes.

gunan committed:
Upgrade ci docker images to cuda 10.

These really should be gone already...

PiperOrigin-RevId: 227914258

aselle committed:
Internal change.

PiperOrigin-RevId: 227920137

aselle committed:
Merge pull request #24711 from aselle/cherrypick-20180104

Additional release cherrypicks - cuda 10 docker image & nomac tag handling.

aselle committed:
Try to make resize bilinear test more deterministic.

PiperOrigin-RevId: 228301953

aselle committed:
Use int32_t rather than int32 for compatibility.

PiperOrigin-RevId: 228302002

aselle committed:
Make toco_port trivial initialize-able for test cases.

Problem is toco_port indirects a way of initializing google.
One impl has a static way of finding out if it is initialized (internal).
One impl has a non-static way to find out.

In the non-static way, test cases won't work since we initialize the tensorflow
google emulation elsewhere (gtest override)

PiperOrigin-RevId: 228306710

aselle committed:
Merged commit includes the following changes:
228181904  by A. Unique TensorFlower<gardener@tensorflow.org>:

    Fix py2 and py3 of gen_git_version.py to fix 1.13 release.

    Byteify function to handle py2 and py3.

--

PiperOrigin-RevId: 228190834

aselle committed:
Merge pull request #24766 from aselle/cherrypick-20180108

Cherrypick additional release 1.13 build fixes.

aselle committed:
Give bytes constructor the appropriate utf-8 argument.

PiperOrigin-RevId: 228342070

jdduke committed:
Fix linkage for several lite/testing tests

Use the tf_cc_test rule for targets that have TensorFlow deps.

PiperOrigin-RevId: 228344897

aselle committed:
Use correct _t integer types to fix mac build.

PiperOrigin-RevId: 228347592

aselle committed:
Merge pull request #24776 from aselle/cherrypick-20180108b

Additional test fixes for 1.13

tensorflower-gardener committed:
Fixing flaky test in quantile regression

PiperOrigin-RevId: 228376536

jdduke committed:
Fix tf_driver_test

Use the proper path for the tested model.

PiperOrigin-RevId: 228415121

jdduke committed:
Fix tf_driver_test

Use the proper path for the tested model.

PiperOrigin-RevId: 228415121

tfboyd committed:
Installs TensorRT

aselle committed:
Merge pull request #24797 from aselle/cherrypick-20180109a

Fix TENSORRT docker image and flaky test.

aselle committed:
Merge pull request #24749 from penpornk/cherrypicks_I3PIP

r1.13-rc0 cherry-pick request: Fix bug in FuseTransposeMklOpTranspose

jdduke committed:
Automated rollback of commit 9f3acf3d89a4ef5f9313a6e2cb2d81e03177e8ef

PiperOrigin-RevId: 228519835

aselle committed:
Move explicit template instantiations to namespace scope.

Fixes compile error on mac os on attached bug causing 1.13
build problems.

PiperOrigin-RevId: 228522576

aselle committed:
Merge pull request #24810 from aselle/cherrypick-20180109d

Various build fixes for 1.13

aaroey committed:
Reduce max_workspace_size_bytes so the GPU memory allocation doesn't exceed the
limit.

PiperOrigin-RevId: 228555498

aselle committed:
Merge pull request #24799 from aselle/cherrypick-20180109c

Avoid grappler bug by restricting to CPU

jdduke committed:
Merging

karimnosseir committed:
Add the data file to the data section in BUILD.

PiperOrigin-RevId: 228782621

aaroey committed:
Put all calibration resources into resource manager. This fixes a race condition
triggered in int8 mode: during calibration, when two sess.run()s fetch two
different sets of ops and both of them trigger the computation of the
TRTEngineOp, two different TRTEngineOp will be created, and dev_tensors_ will
only be allocated once by the first op by the device's resource manager via
LookupOrCreate(). As a result, the second TRTEngineOp will have a empty
dev_tensors_ vector which will result in std::out_of_range error when the op
tries to access it later.

We should consider marking TRTEngineOp as stateful, the problem is it uses TF
functions which has to be stateless, but we can use function library as key to
cache them.

PiperOrigin-RevId: 228816949

aselle committed:
Print disk usage after building stuff.

PiperOrigin-RevId: 228934277

penpornk committed:
Fix --config=mkl performance regression: Prevent grappler from generating a _FusedConv2D node that doesn't have a corresponding TF-MKL node. For example, MKL graph rewriter doesn't support _FusedConv2D with BatchNorm yet so any Conv2D that gets fused with BatchNorm will miss using MKL-DNN's convolutions.

PiperOrigin-RevId: 228939593

aaroey committed:
Add a mutex for each (ICudaEngine, IExecutionContext) pair since some of their
operations are not thread safe.

This fixed one of the issues mentioned in
#23853

PiperOrigin-RevId: 228947504

aselle committed:
Debug disk usage of parameterized build.

PiperOrigin-RevId: 228960346

aaroey committed:
Increase the tolerance value for quantization_mnist_test.

PiperOrigin-RevId: 228980452

aselle committed:
Merge pull request #24880 from aselle/cherrypick-20190113

Cherrypick disk space debugging changes.

dmitrievanthony committed:
Fix Ignite Dataset (V2 API).

dmitrievanthony committed:
Fix layout of README.md in ignite module.

dmitrievanthony committed:
Update Ignite tests

dmitrievanthony committed:
Fix Kafka Dataset (V2 API).

aselle committed:
Merge pull request #24900 from jdduke/cherrypicks_ONNPG

Cherry pick of "Fix tensor_utils_test"

yongtang committed:
Update kinesis dataset to match v2 API

Since DatasetSource now requires varient_tensor to be passed into constructor,
the kinesis data will need to be updated to fix the incompatibility issue.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

aselle committed:
Merge pull request #24907 from jdduke/cherrypicks_V6WMP

Add the data file to the data section in BUILD.

aselle committed:
Merge pull request #24902 from jdduke/cherrypicks_CWIJ1

Cherrypick of "Fix tf_driver_test"

aselle committed:
More disk space debugging messages.

aselle committed:
Merge pull request #24908 from aselle/disk-space-debug

More disk space debugging messages.

aselle committed:
Merge pull request #24909 from aselle/cherrypick-20190114

Cherrypick cl/20190114

jdduke committed:
Disable InterpreterFlexTest

This test currently only works with --config=monolithic. Disable for
now until the select .so supports execution against the shared TF lib.

PiperOrigin-RevId: 229269278

aaroey committed:
Set the data type of the IConstantLayer output correctly.

PiperOrigin-RevId: 229285591

aselle committed:
Merge pull request #24911 from jdduke/cherrypicks_CUFZ8

Disable InterpreterFlexTest

jdduke committed:
Fix several TFLite tests

PiperOrigin-RevId: 229288943

This is a partial cherry-pick because it includes fixes for tests that
didn't exist previously.

tensorflower-gardener committed:
Fix transpose_conv

Use the proper filter height/width dimensions.

PiperOrigin-RevId: 229294726

aselle committed:
Try to fix gen_git_source.py some more.

aselle committed:
Merge pull request #24939 from aselle/cherrypick-20190115

Cherrypick 20190115 (fixes some tflite buidls)

aselle committed:
Merge pull request #24937 from aselle/gen_git_source_fix

Try to fix gen_git_source.py some more.

aselle committed:
Avoid using a temp directory with a space to workaround bazel errors.

PiperOrigin-RevId: 229422438

jdduke committed:
Fix arena_planner_test

Use proper tf_cc_test rule.

PiperOrigin-RevId: 229432697

aselle committed:
Merge pull request #24862 from aaroey/r1.13

r1.13-rc0 cherry-pick request: fix races in TF-TRT integration

aselle committed:
Merge pull request #24956 from aselle/20190115b

Cherry pick windows build fix and tflite test fixes.

aselle committed:
Merge pull request #24941 from aaroey/fix_quantization_mnist_test_for_1.13

Increase the tolerance value for quantization_mnist_test.

aselle committed:
Add tf_cc_test to build file to fix sanity check.

jdduke committed:
Fix more TFLite tests and binaries

Use the tf_cc_binary where appropriate. Also fix an issue
with schema generation and blacklist lstm zip tests temporarily.

PiperOrigin-RevId: 229596167

aselle committed:
Merge pull request #24973 from aselle/fix-sanity-tf-cc

Add tf_cc_test to build file to fix sanity check.

aselle committed:
Merge pull request #24975 from aselle/cp-20190116

Cherry pick fix to tflite tests.

aselle committed:
Fix merge conflict.

aselle committed:
Merge pull request #24861 from penpornk/cherrypicks_1VHQW

r1.13-rc0 cherry-pick request: Fix --config=mkl performance regression

caisq committed:
Disable testing of user-defined GPU kernel under eager mode

PiperOrigin-RevId: 229644310

allenlavoie committed:
Always call PyObject_ClearWeakRefs

This seems to work in Python 2 and be necessary for correctness in Python 3. weakreflist is not set and yet there are weakrefs which need to be cleared.

Oddly it's only the weak key dictionary that fails in Python 3 before removing the conditional; using weakref.ref on its own seems fine.

PiperOrigin-RevId: 229645639

aselle committed:
Merge pull request #25004 from aselle/cp-20190117

Cherry pick additional test failures.

aselle committed:
Disable InterpreterTest.java for now and fix missing include.

PiperOrigin-RevId: 229813499

aselle committed:
Make tolerance more reasonable on _testOptimizersCompat...

PiperOrigin-RevId: 229817236

aselle committed:
Further make tolerance larger for cudnn comparison

PiperOrigin-RevId: 229818348

aselle committed:
Make tolerance larger for cudnn comparison

PiperOrigin-RevId: 229818348

aselle committed:
Disable broken tests for 1.13

PiperOrigin-RevId: 229820397

aselle committed:
Merge pull request #25014 from aselle/20190117bc

Fix several more r1.13 related test failures.

aselle committed:
Disable jni on mac since mac doesn't use gnu linker

This does not preclude building Android Java JNI since that uses a different toolchain that does support version-script

PiperOrigin-RevId: 229946385

aselle committed:
Merge pull request #25032 from aselle/20190117b

Disable jni tests on mac (x86_64 toolchain not compatible with our linker script)

aselle committed:
Added release notes for 1.13 release

aselle committed:
Updated release notes

jtkeeling committed:
Add bounds checking when looking at the stack in deprecation messages.

This was breaking integration with R, fixes #25067.

PiperOrigin-RevId: 230431304

aselle committed:
Remove spurious whitespace changes.

aselle committed:
Add comment about Apache Ignite File System note in release notes.

aselle committed:
Merge pull request #25084 from aselle/release-notes-1.13

Added release notes for 1.13 release

aselle committed:
Updated release notes to additional 2 herry-picks

aselle committed:
Merge pull request #25125 from aselle/cp-20190123

Cherry pick openssl fix and R fix and update release notes.

Commit:
Update RELEASE.md

dmitrievanthony committed:
Remove import of tensorflow.

mihaimaruseac committed:
Add Estimator release notes to 1.13

This is in preparation to releasing Estimator 1.13, as per the instructions

mihaimaruseac committed:
Minor changes to make estimator notes consistent

Could be squashed

Commit:
Update Estimator version in setup,py

This needs to be cherry-picked to TF r1.13 branch.
There now exists a tensorflow_estimator==1.13.0rc0 release on pypi.

PiperOrigin-RevId: 230987863

tensorflower-gardener committed:
Fix estimator dependency.

This needs to be cherry-picked to TF r1.13 branch.

PiperOrigin-RevId: 231209093

Commit:
Change estimator dep to >1.13rc0.

PiperOrigin-RevId: 231246464

aselle committed:
Merge pull request #25142 from tensorflow/rachellim-patch-1

Update RELEASE.md

aselle committed:
Merge pull request #25266 from aselle/cp-20190128

Fix TransposeConv op kernel in TensorFlow Lite

jsimsa committed:
[tf.data] Avoid printing a spurious warning for `tf.compat.v1.data.make_one_shot_iterator` and `tf.compat.v1.data.make_initializable_iterator`.

PiperOrigin-RevId: 231281365

aselle committed:
Merge pull request #25268 from aselle/cp-20190128b

Cherrypick fixes related to estimator dependency

aselle committed:
Merge pull request #25286 from stephanwlee/r1.13

[CP request] Fix GFile tf_export for v2

aselle committed:
Merge pull request #25290 from aselle/cp-20190124

Merge ignite fixes for 1.13

ashahba committed:
Set default TF_BUILD_VERSION to r1.13 on MKL images

mihaimaruseac committed:
Prevent null pointer dereference in decode_gif.

PiperOrigin-RevId: 231841542

aselle committed:
Merge pull request #25399 from mihaimaruseac/cherrypicks_U3TI0

[v1.13][CP Request] Prevent null pointer dereference in decode_gif.

aselle committed:
Merge pull request #25267 from jsimsa/cherrypicks_HTVMS

<release 13>-<rc1> cherry-pick request: Removing spurious warning in tf.data.

aselle committed:
Merge pull request #25309 from ashahba/ashahba/r1.13-mkl

Set default TF_BUILD_VERSION to r1.13 on MKL images

aselle committed:
Merge pull request #25028 from yongtang/r1.13

[r1.13] Update kinesis dataset to match v2 API

aselle committed:
Update version to 1.13.0-rc1

aselle committed:
Merge pull request #25489 from tensorflow/aselle-cp-20190204

Change version to 1.13.0-rc1

aselle committed:
Revert "[tf.data] Avoid printing a spurious warning for `tf.compat.v1.data.make_one_shot_iterator` and `tf.compat.v1.data.make_initializable_iterator`."

This reverts commit 75dd907a8d6848c4517421317cb2c3bec8fdd8e0.

aselle committed:
Merge pull request #25497 from tensorflow/aselle-cp-20190204b

Revert dataset warning fix as it breaks many tests.

fchollet committed:
Fix applications integration test to be compatible with latest release of keras_applications

aselle committed:
Merge pull request #25498 from fchollet/r1.13

Fix applications integration test to be compatible with latest release of Keras-Applications

Commit:
Skip testNonMatchingVariableCreation.

PiperOrigin-RevId: 232547857

aselle committed:
Disable all toco batch mul tests

aaroey committed:
Update RELEASE.md

Add TensorRT 5.0 support in release notes.

aselle committed:
Merge pull request #25562 from tensorflow/aaroey-1.13-release-note

Update RELEASE.md

Commit:
Adding Python3.7 to release notes.

wookayin committed:
Mention 'num_parallel_calls' bugfix into 1.13 release note

Mention Bug #19945 in the release note of TF 1.13.

aselle committed:
Merge pull request #25671 from aselle/cp-20190211

Cherry pick fix to 1.13 for python 3

aselle committed:
Revert "Handle the case that CancellationManager is null in eager mode"

This reverts commit 5bb0553a1b3bd580bd1502ab6b339ca3a1f0b5df.

aselle committed:
Revert "Raise an error when loop execution is cancelled"

This reverts commit fc5392dedec126f988788a597edb55021fb07b60.

aselle committed:
Revert "Add a python test for while_loop timeout"

This reverts commit c0b128c45396560f26769a293525c60f76850a3f.

aselle committed:
Revert "Add CancellationManager to LoopCondOp"

This reverts commit 59ebe545b3385c4c36d3b1602671d109e44ea38c.

aselle committed:
Remove deprecation warning on tf.data.v1 functions

This is for 1.13 only and should not be merged back into master.
This is a replacement for PR#25578 on 1.13 branch.

aselle committed:
Merge pull request #25696 from aselle/cp-20190212

Revert timeout for while loop

aselle committed:
Merge pull request #25698 from aselle/cp-20190212c

Remove deprecation warning on tf.data.v1 functions

nhasabni committed:
[Intel MKL] Fix incorrect way to dump optimized graph

This PR fixes issue 25674. It simply adds a check to ensure that
a graph is valid before dumping it.

aselle committed:
Update version to 1.1.13-rc2

aselle committed:
Merge pull request #25633 from wookayin/patch-1

Mention 'num_parallel_calls' bugfix into 1.13 release note

aselle committed:
Merge pull request #25725 from aselle/cp-20190213

Prepare for rc2 by updating version & fixing tolerance in contrib cudnn test

aselle committed:
Merge pull request #25770 from aselle/cp-20190214

[Intel MKL] Fix incorrect way to dump optimized graph

aselle committed:
Merge pull request #25769 from aselle/cp-20190214b

Fix regression in XLA in TensorFlow 1.13 RC0, RC1

yhliang2018 committed:
Fix serialization naming for BatchNorm.

PiperOrigin-RevId: 234726512

gunan committed:
Update package version to 1.13.0.

frankchn commented:
@rjpower @xiejw PTAL? Thanks!

rjpower commented on tensorflow/contrib/tpu/python/tpu/keras_support.py:
```diff
@@ -2145,20 +2145,23 @@ def load_weights(self, filepath, by_name=False):
-      input_shapes = [layer.input_shape]
-    else:
-      input_shapes = layer.input_shape
+    for node_idx in range(len(layer._inbound_nodes)):
```
It seems like we want to accumulate the input and output shapes here. This appears to take only the last shape into account, and assumes the number of inputs and outputs are the same:

```
input_shapes = []
output_shapes =[]

for node_idx in range(...):
  input_shapes.append(layer.get_input_shape_at(node_idx))

for node_idx in range(...):
  output_shapes.append(layer.get_output_shape_at(node_idx)
```

For testing, I've found it is often easiest to use the TPU enable version of Colab, replace/sed the keras_support file and restart the notebook:

```
import tensorflow as tf
from tensorflow.contrib.tpu.python.tpu import keras_support
keras_support.__file__

with open('/usr/local/lib/python3.6/dist-packages/tensorflow/contrib/tpu/python/tpu/keras_support.py', 'w') as f:
  f.write(NEW_CONTENT)
```

ctessum replied:
My understanding is that there are 2 concepts here, multiple nodes and multiple inputs/outputs, where multiple nodes means that the same layer (with the same weights) is being used in more than one location in the graph. I was assuming that everything was working correctly except for the fact that this code was assuming only one node per layer, so that's what this change was aimed at, but I will need to test it to confirm.

I ran into problems testing it on colab because I tried to just replace the whole file, but then it no longer compiles with tf 1.13rc2. It works with tf-nightly, but there appears to be a regression in tf-nightly that causes my model to no longer compile.

I'll keep working on it.

rjpower reviewed (changes requested):
Thanks for taking this on!

nfelt committed:
Update tensorboard dependency to 1.13.x

TensorBoard release: https://pypi.org/project/tensorboard/1.13.0/

PiperOrigin-RevId: 235563447

gunan committed:
Bump TF version to 1.13.1

gunan committed:
Merge pull request #26101 from gunan/r1.13

Bump TF version to 1.13.1

ctessum committed:
TPU: keras support for multiple layer nodes

Hopefully fixes #23659.

ctessum commented:
Ack! I rebased it onto the 1.13.1 release so that it would be less of a moving target for copying into colab, but there were unintended consequences! I will rebase it onto master once I'm done testing.

ctessum commented:
So when I implement this with my model in colab, the error no longer occurs, but `tf.contrib.tpu.keras_to_tpu_model` stalls with the "running" symbol seemingly forever (I've been waiting for an hour now). I could try to test it with a simpler example, but because it doesn't fix my use case I think I am going to close this and wait for a more permanent solution, because my understanding is that this function is slated to be replaced anyway.
